packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -562,6 +585,9 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
             // returned in packagePath().
             Object.freeze(this.pkg);
 
+            // Update the used routes of the channel in case GC was run before we were realized.
+            this.updateChannelUsedRoutes();
```

Comment:
I believe the intention is for data store to be able to reason about non-referenced DDSs and be able to purge them.
And I'd guess same for sub-DDS level.
Have you considered the impact of calling it at random times (relative to when summarization / snapshot runs)?
I.e. I'd assume that something like at-mentions data store would not be able to use results of this call if it happens outside of summary processing, as undo/redo could reinsert content that points to same route (in case of at-mentions, it's just a key in DDS - the same key gets reinserted, but deleting it after undo as unused would be incorrect).

Reply:
As per our discussion offline, it is currently going to be difficult to handle the scenario where data stores start using the result of this call to delete stuff and undo / redo happens. Even if this is tied to summary, it is hard to get this right because of the ordering of the delete op and undo / redo op.
So maybe these data stores should not delete things but only exclude them from search results to begin with.